semgrep-core/src/tainting/Dataflow_tainting.ml

```diff
@@ -58,6 +58,13 @@ type overlap = float
 type propagator_id = var
 type propagator_from = propagator_id
 type propagator_to = propagator_id
+type taint_info = Tainted of Taints.t | MarkedClean
```

Comment:
Ideally we didn't need `MarkedClean` but I don't have any better proposal right now, I'll think on it.

Reply:
I agree that it feels not so elegant. But I couldn't come up with a better solution.